app/models/subnet/ipv4.rb

```diff
@@ -0,0 +1,38 @@
+require 'socket'
+
+class Subnet::Ipv4 < Subnet
+  has_many :interfaces, :class_name => 'Nic::Base', :foreign_key => :subnet_id
+  has_many :primary_interfaces, -> { where(:primary => true) }, :class_name => 'Nic::Base', :foreign_key => :subnet_id
+  has_many :hosts, :through => :interfaces
+  has_many :primary_hosts, :through => :primary_interfaces, :source => :host
```

Comment:
If I don't move them (read: leave them in the Subnet class), rails complains.
Is there anything I can do about this?

Apparently, you have to define the "has_many" before the "has_many through"....

```
ActiveRecord::HasManyThroughAssociationNotFoundError: Could not find the association :interfaces in model Subnet
```